compose/config/interpolation.py

```diff
@@ -26,9 +29,46 @@ def process_item(name, config_dict):
     )
 
 
-def interpolate_value(name, config_key, value, section, mapping):
+def _cast_interpolated(interpolated_value, expected_type):
+
+    if expected_type == "string":
+        return None
+    elif expected_type == "number":
+        try:
+            return int(interpolated_value)
+        except ValueError:
+            return None
+    elif expected_type == "boolean":
+        if "true" == interpolated_value.lower():
+            return True
+        if "false" == interpolated_value.lower():
+            return False
+        return None
```

Comment:
In the case where no casting happens, I think this function should just return the original value. That way there is no need to check for `None`. It can just return normally. It should simplify the callers logic a bit.